src/app/product-edit/product-edit.component.html

```diff
@@ -1,21 +1,33 @@
-<h1 *ngIf="product">Edit: {{product.name}}</h1>
-<form [formGroup]="editForm" (ngSubmit)="onSubmit()">
+<div>
```

Comment:
I'd like you to extract the product form into a separate component and pass the form object as input. You essentially have the same thing duplicated in both edit and add.